File: php/collatz-conjecture/CollatzConjecture.php
<?php

declare(strict_types=1);

function steps(int $n): int
{
    if ($n <= 0) {
        throw new InvalidArgumentException('Only positive numbers are allowed');
    }
    $step = 0;
    while ($n > 1) {
        $n = $n % 2 === 0 ? $n / 2 : 3 * $n + 1;
        ++$step;
    }
    return $step;
}

File: euphoria/collatz-conjecture/collatz-conjecture.ex
public function steps(integer n)
    integer cnt = 0
    if n < 1 then return "Only positive numbers are allowed" end if
    while n > 1 do
        if remainder(n, 2) = 0 then
            n /= 2
        else
            n = n * 3 + 1
        end if
        cnt += 1
    end while
    return cnt
end function
